### PR TITLE
Update README and CONTRIBUTING docs

### PR DESCRIPTION
**Pull request by mperrotti**
I've updated the README.md and CONTRIBUTING.md with more accurate information about the current state of ui-kit.

I also changed the default component directory structure to be flat. The `components`, `tests`, and `stories` directories just feel noisy to me.

## Testing

Read the updated README.md and CONTRIBUTING.md and give feedback on how they could be improved.

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [ ] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [ ] Info for applicable sections above is provided

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,110 +62,133 @@ info
 
 After this there is a web server started at the address http://localhost:6006 which is hosting storybook. This is our documentation from the local repository. Anything you change will reflect there directly.
 
-Change what you want to change. After you are happy with the changes you want to run `npm test` and `npm run lint`.
+After you're happy with your channges, create a commit by using `git commit` or using [git cz](http://commitizen.github.io/cz-cli/).
```
Comment (pierrebeitz) [inline CONTRIBUTING.md:65]: now that's a cool tool! 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -66,110 +62,133 @@ info
...
+For example, a component for a card that may contain any content would be a good candidate for ui-kit. A card that specifically renders data specific to an object in an napp is not.
 
-* Link  
-  We add a link to that component to the documentation. This is not a favorable way.
-* Wrap  
-  We are creating our own component but in that component we are utilising a 3rd party component, this solution is the favourable solution for the `UI-kit` as with that solution we are adding consistency to the 3rd party components and there is a single point to change the usage of that component if we decide to update the component or replace the component with our own implementation or the other implementation.
-* Fork  
-   We should do that as a last resort if we do not see any other solution.
+**Does it fit in with a "theme" of other components in the UI Kit?**
+For example, if we have a toast and an inline alert box, a page banner would make sense to exist in the ui-kit
```
Comment (pierrebeitz) [inline CONTRIBUTING.md:89]: i have to admit that i don't immediately see the correlation of `toast`, `alert box` and `page banner`. 🤔 

but that's in no way blocking....

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -66,110 +62,133 @@ info
...
 
-Going from link via wrap to fork increases the control over component but also increases the effort need to update the component. A wrap is a sweet spot between consistency and functionality vs control and effort.
+**Will side effects of adding this component help us move the design system forward?**
+For example, adding a toast component forces us to think about visual indication of alert status (error, success, warning, etc.), z-index stacking context, accessibility guidelines, and how we animate a piece of UI in and out of the viewport.
 
-### Moving A Component From `dcos/dcos-ui` To The `ui-kit`
+**Is now the right time to add this to UI Kit?**
+If adding this component right now prevent us from working on more important work, maybe it should be built in dcos-ui, and we’ll follow-up about adding to UI Kit at the end of the release iteration.
+If I can’t get this added to UI Kit in time for it to be used in dcos-ui to resolve your feature/task/bug, build in dcos-ui, and we’ll follow-up about adding to UI Kit at the end of the release iteration.
```
Comment (pierrebeitz) [inline CONTRIBUTING.md:99]: i think we don't need to mention dcos-ui in here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -66,110 +62,133 @@ info
...
-### Creating a Reusable Component
+## Maintaining Reusable Components
+
+### Rigid vs. Flexible
+
+Generally, a component should rather be rigid than flexible. That means that we rather create more components then creating one which has a high complexity. Although a flexible component has a higher rate of reusability in more contexts the complexity increases a lot more. Which is an easy way to introduce a lot of errors to a component, that is why we prefer to have rigid components. A rigid component is simple, easy to understand, easy to test, less code and also improves the consistency.
+
+> It's always easy to add flexibility later, it's nearly impossible to remove flexibility.
+
+> -- Random Person on the Internet
```
Comment (pierrebeitz) [inline CONTRIBUTING.md:128]: 😂

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -183,7 +202,9 @@ It's important to keep components as simple as possible, add logic to a componen
 
 #### Directory Structure
 
-Components get grouped into `packages`. The Strucutre looks like this:
+Each component has a directory in `packages/`.
+
+Start with a flat directory. Then, if a component gets complex and has a lot of files, add directories like `components`, `tests`, and `stories` if they add clarity.
```
Comment (pierrebeitz) [inline CONTRIBUTING.md:207]: # ❤️ ❤️ ❤️

**Review by pierrebeitz — DISMISSED** [review]
i just skimmed through the text changes and did not try the template generation nor have i verified that all links in `CONTRIBUTING.md`s TOC work.

i think that dcos-ui won't have much impact on ui-kit in the future and we can safely remove references to it.

i like this change a lot!

File: CONTRIBUTING.md (hunk 3 above)
Comment (mperrotti) [reply CONTRIBUTING.md:99]: You're absolutely right, my mistake. I copied that paragraph from something I wrote in 2018 😅

File: CONTRIBUTING.md (hunk 4 above)
Comment (mperrotti) [reply CONTRIBUTING.md:128]: This is from Philipp 😛

**Pull request merged**